live-build/misc/ansible-roles/appliance-build.minimal-common/tasks/main.yml

```diff
@@ -363,14 +363,28 @@
     regexp: '^GRUB_CMDLINE_LINUX_DEFAULT='
     line: 'GRUB_CMDLINE_LINUX_DEFAULT="$GRUB_CMDLINE_LINUX_DEFAULT crashkernel=1024M-:512M"'
 
-# Panic on ZFS assertion failures
 - lineinfile:
     path: /etc/modprobe.d/zfs.conf
     create: yes
-    line: 'options spl spl_panic_halt=1'
+    line: "{{ item }}"
   # Because the spl module is loaded from the initramfs when using zfs on root,
   # we need to rebuild the initramfs to account for the modprobe.d change.
   notify: update initramfs
+  with_items:
+    #
+    # Panic on ZFS assertion failures; otherwise it's possible for these
+    # assertion failures to only halt the thread that failed the assert,
+    # which can then put the entire system into a "hung" state, since
+    # we're running a rootfs on ZFS.
+    #
+    - "options spl spl_panic_halt=1"
+    #
+    # We increase the max memory limit for ZFS channel programs to 512MB
+    # to give us a wide buffer for development of channel programs. We
+    # were using upwards of 70MB during internal testing, which is too
+    # much with the default limit of 100MB.
+    #
+    - "options zfs zfs_lua_max_memlimit=536870912"
```

Comment:
We can change tunables either by modifying the conf file (as you're doing here) or by changing the source code (of github/delphix/zfs).  We've done the latter for some other tunables.  How should we determine which mechanism to use?

Reply:
I’m OK with either approach. I asked on Slack, and was directed to this approach. 

@ahrens let me know if you’d rather us make this change in the ZFS code, instead.

Reply:
cc @gwilson

Reply:
We've made these tunable changes in delphix/zfs that are not present upstream (zfsonlinux/zfs).  We should make all the tunable changes in one place, or have some criteria for determining where to make them.

```
+int zfs_dmu_offset_next_sync = 1
+int zfs_mg_noalloc_threshold = 5;
+int spa_asize_inflation = SPA_DVAS_PER_BP + 1;
+uint32_t zfs_vdev_sync_read_min_active = 35;
+uint32_t zfs_vdev_sync_read_max_active = 35;
+uint32_t zfs_vdev_sync_write_min_active = 35;
+uint32_t zfs_vdev_sync_write_max_active = 35;
+uint32_t zfs_vdev_async_read_max_active = 10;
+uint32_t zfs_vdev_async_write_min_active = 1;
+int zfs_vdev_aggregation_limit = SPA_OLD_MAXBLOCKSIZE;
```

Reply:
I also started to make these change in `delphix-config` so we should figure out where the final home for these changes should be.

Reply:
Are you suggesting we should opt to go the route of using this config file as a general rule of thumb?

If so, I can open another PR to pull in those settings as well. And then another PR to zfs to revert back to defaults.

Personally, I like this config file approach since it makes it clear which tunables we change, but I’d be OK going the route of making the change in our ZFS too.

Which approach do you prefer? I can’t quite tell so far..

Reply:
@gwilson the problem with doing this in delphix-config, is it then requires a reboot (and rebuilding the initramfs) to take effect.